uarray/_uarray_dispatch.cxx

```diff
@@ -227,6 +231,46 @@ PyObject * register_backend(PyObject * /*self*/, PyObject * args)
   Py_RETURN_NONE;
 }
 
+void clear_single(std::string domain, bool registered, bool global)
+{
+  if (global_domain_map.find(domain) == global_domain_map.end())
```

Comment:
Save the result from the first lookup and then you don't need to do repeated lookups.